src/utilities/calculateWinner.tsx

```diff
@@ -0,0 +1,27 @@
+/**
+ * Calculate the game winner
+ * @param {*} squares
+ */
+
+export const calculateWinner = (squares: Array<string>): WinnerState => {
+  const lines: Array<Array<number>> = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+  for (let i = 0; i < lines.length; i++) {
```

Comment:
you could use an array method to make this more declarative, probably a find. Not a big deal though